src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java

```diff
@@ -101,6 +101,7 @@ public class InnerAssignmentCheck
             TokenTypes.RESOURCE_SPECIFICATION,
         },
         {TokenTypes.EXPR, TokenTypes.LAMBDA},
+        {TokenTypes.EXPR, TokenTypes.SWITCH_RULE},
```

Comment:
I don't see it being asked anywhere, but why are we damaging lambda expression by removing them from this list?

https://github.com/checkstyle/checkstyle/commit/f798a441fcd58c890ca44b0cf276bca5752b10e2
https://github.com/checkstyle/checkstyle/issues/377